### PR TITLE
Optimize EvmStack

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Load and endianness swap in one instruction
- Tail call for Int256
- Move exceptions out of flow
- Move tracing loops out of flow

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by LukaszRozmej** [comment]
Running hive tests:
https://github.com/NethermindEth/nethermind/actions/runs/4333660864
https://github.com/NethermindEth/nethermind/actions/runs/4333663830

**Review by LukaszRozmej — APPROVED** [review]

**Comment by rubo** [comment]
> These errors from this change? 🤔

Nope. They are just flaky.

File: src/Nethermind/Nethermind.Evm/EvmStack.cs [hunk 1]
```diff
@@ -181,51 +185,53 @@ public void PushUInt256(in UInt256 value)
...
                 Metrics.EvmExceptions++;
-                throw new EvmStackOverflowException();
+                ThrowEvmStackOverflowException();
             }
         }
 
         public void PushSignedInt256(in Int256.Int256 value)
         {
-            PushUInt256((UInt256)value);
+            // tail call into UInt256
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Evm/EvmStack.cs:217]: QQ: Can you elaborate what you mean by "tail call" here? Does .net have some specific optimication in this case?

**Review by marcindsobczak — APPROVED** [review]
Code looks fine, hive consensus tests are green, if engine tests would pass as well (or we will have same false-positive fails from master) I will feel comfortable with merging it

File: src/Nethermind/Nethermind.Evm/EvmStack.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Evm/EvmStack.cs:217]: No stack to clean up; return type and parameters match; so in the asm these method just turns into a single jump to the other methods
```asm
; Method Nethermind.Evm.EvmStack:PushSignedInt256(byref):this
G_M000_IG01:

G_M000_IG02:
       FF25D23F1400         tail.jmp [Nethermind.Evm.EvmStack:PushUInt256(byref):this]
; Total bytes of code: 6
```
```asm
; Method Nethermind.Evm.EvmStack:PopSignedInt256(byref):this
G_M000_IG01:

G_M000_IG02:
       FF2532401400         tail.jmp [Nethermind.Evm.EvmStack:PopUInt256(byref):this]
; Total bytes of code: 6

```

**Pull request merged**